python/Ryu.py

```diff
@@ -273,31 +282,247 @@ def changeSwitchForwardingState(self, dpid=None, header=None, actions=None, comm
+            del obj['tcp_src']
+            
+
+        if(header.has_key('tcp_dst')):
+            obj['tcp_dst'] = header['tcp_dst']
+            obj['ip_proto'] = 6
+        else:
+            del obj['tcp_dst'] 
+            
+
+        if(header.has_key('udp_src')):
+            obj['udp_src'] = header['udp_src']
+            obj['ip_proto'] = 17
+        else:
+            del obj['udp_src']
+            
+
+        if(header.has_key('udp_dst')):
+            obj['udp_dst'] = header['udp_dst']
+            obj['ip_proto'] = 17
+        else:
+            del obj['udp_dst']
+            
+                
+        if obj['eth_type'] is None:
+            del obj['eth_type']
+
+        if obj['ip_proto'] is None:
+            del obj['ip_proto']
+        
+        match = parser.OFPMatch(**obj)
+        of_actions = []
+        for action in actions:
+            if(action['type'] == "output"):
+                of_actions.append(parser.OFPActionOutput(port     = int(action['port']),
+                                                         max_len  = 0
+                                                         ))
+
+        if header.has_key('dl_type'):
+            del header["dl_type"]
+
+        if obj.has_key('eth_type'):
+            header['eth_type'] = obj['eth_type']
+
+        dpid =  "%016x" % dp.id
+        idle = {}
+        hard =  {}
+        now = time.time()
+        
+        if int(idle_timeout) != 0:
+            timeout = now + int(idle_timeout)
+            idle   =      {'timeout': timeout,
+                           'dpid': dpid,
+                           'domain' : domain,
+                           'idle_timeout': int(idle_timeout),
+                           'pkt_count': 0,
+                           'header': header,
+                           'actions': of_actions,
+                           'priority': priority,
+                           'command': "ADD"}
+
+        if int(hard_timeout) != 0:
+            timeout = now + int(hard_timeout)
+            hard =        {'timeout': timeout,
+                           'dpid': dpid,
+                           'domain' : domain,
+                           'idle_timeout': int(hard_timeout),
+                           'pkt_count': 0,
+                           'header': header,
+                           'actions': of_actions,
+                           'priority': priority,
+                           'command': "ADD"}
+            
+        self.logger.debug("Actions: " + str(of_actions))
+        flags = 0
+        if(command == "ADD"):
+            command = ofp.OFPFC_ADD
+            flags = ofp.OFPFF_SEND_FLOW_REM
+            self.api.pushTimeouts(idle, hard)
+            self.api.pushFlows(dpid, domain, header, actions, priority)
+        elif(command == "DELETE_STRICT"):
+            command = ofp.OFPFC_DELETE_STRICT
+            self.api.remove_flow(dpid, domain, header, priority)
+        else:
+            command = ofp.OFPFC_DELETE
```

Comment:
Why would we only want to delete instead of delete_strict?  Seems like we could accidentally delete a lot of flows?